core/primitives/src/network.rs

```diff
@@ -9,13 +10,29 @@ use near_crypto::{KeyType, PublicKey, SecretKey, Signature};
 use crate::hash::{hash, CryptoHash};
 use crate::types::{AccountId, EpochId};
 
+/// Peer id is the public key.
+#[derive(
+    BorshSerialize, BorshDeserialize, Clone, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
+)]
+pub struct PeerId(Arc<PeerIdInner>);
+
 /// Peer id is the public key.
 #[derive(
     BorshSerialize, BorshDeserialize, Clone, Eq, PartialOrd, Ord, Serialize, Deserialize, Hash,
 )]
-pub struct PeerId(pub PublicKey);
+pub struct PeerIdInner(pub PublicKey);
```

Comment:
Doesn't need to be `pub`

Reply:
@pmnoxx I'm thinking about rewriting `PeerId` inner to have same format at `Edge`. That is that most methods will be in `PeerId`.